extensions/binding/org.eclipse.smarthome.binding.fsinternetradio/src/main/java/org/eclipse/smarthome/binding/fsinternetradio/internal/FSInternetRadioDiscoveryParticipant.java

```diff
@@ -0,0 +1,165 @@
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.jupnp.model.meta.DeviceDetails;
+import org.jupnp.model.meta.ManufacturerDetails;
+import org.jupnp.model.meta.ModelDetails;
+import org.jupnp.model.meta.RemoteDevice;
+import org.jupnp.model.meta.RemoteDeviceIdentity;
+
+/**
+ * This is the discovery service for internet radios based on the fontier silicon chipset. Unfortunately, it is not
+ * easily possible to detect from the upnp information which devices are supported. So currently, discovery only works
+ * for medion internet radios. {@link FSInternetRadioDiscoveryParticipant#getThingUID(RemoteDevice)} must be extended to
+ * add further supported devices!
+ *
+ * @author Patrick Koenemann - Initial contribution
+ */
+public class FSInternetRadioDiscoveryParticipant implements UpnpDiscoveryParticipant {
+
+    /** Map from UPnP model name to model number for supported radios; filled in static initializer below. */
+    private static final Map<String, Set<String>> SUPPORTED_RADIO_MODELS = new HashMap<String, Set<String>>();
+
+    static {
+        // to allow case-insensitive match: add all values UPPER-CASE!
+
+        // list of medion internet radios taken from: http://internetradio.medion.com/
+        final Set<String> medionRadios = new HashSet<String>();
+        SUPPORTED_RADIO_MODELS.put("MEDION AG", medionRadios);
+        medionRadios.add("MD83813");
+        medionRadios.add("MD84017");
+        medionRadios.add("MD85651");
+        medionRadios.add("MD86062");
+        medionRadios.add("MD86250");
+        medionRadios.add("MD86562");
+        medionRadios.add("MD86672");
+        medionRadios.add("MD86698");
+        medionRadios.add("MD86869");
+        medionRadios.add("MD86891");
+        medionRadios.add("MD86955");
+        medionRadios.add("MD86988");
+        medionRadios.add("MD87090");
+        medionRadios.add("MD87180");
+        medionRadios.add("MD87238");
+        medionRadios.add("MD87267");
+
+        // list of hama internet radios taken from:
+        // https://www.hama.com/action/searchCtrl/search?searchMode=1&q=Internet%20Radio
+        final Set<String> hamaRadios = new HashSet<String>();
+        SUPPORTED_RADIO_MODELS.put("HAMA", hamaRadios);
+        hamaRadios.add("IR100");
+        hamaRadios.add("IR110");
+        hamaRadios.add("IR250");
+        hamaRadios.add("IR320");
+        hamaRadios.add("DIR3000");
+        hamaRadios.add("DIR3100");
+        hamaRadios.add("DIR3110");
+    }
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+        return Collections.singleton(THING_TYPE_RADIO);
+    }
+
+    @Override
+    public DiscoveryResult createResult(RemoteDevice device) {
+        final ThingUID uid = getThingUID(device);
+        if (uid != null) {
+            final Map<String, Object> properties = new HashMap<>(1);
+            final String ip = getIp(device);
+            if (ip != null) {
+                properties.put(CONFIG_PROPERTY_IP, ip);
```

Comment:
it would be great if you would also set "vendor" and "modelId" as properties (if available on the device), see https://www.eclipse.org/smarthome/documentation/development/bindings/thing-definition.html#properties